sdk/storage/azure-storage-blob/azure/storage/blob/_shared/constants.py

```diff
@@ -17,9 +17,10 @@
 # for python 3.5+, there was a change to the definition of the socket timeout (as far as socket.sendall is concerned)
 # The socket timeout is now the maximum total duration to send all data.
 if sys.version_info >= (3, 5):
-    # the timeout to connect is 20 seconds, and the read timeout is 2000 seconds
-    # the 2000 seconds was calculated with: 100MB (max block size)/ 50KB/s (an arbitrarily chosen minimum upload speed)
-    READ_TIMEOUT = 2000
+    # the timeout to connect is 20 seconds, and the read timeout is 80000 seconds
+    # the 80000 seconds was calculated with:
+    # 4000MB (max block size)/ 50KB/s (an arbitrarily chosen minimum upload speed)
+    READ_TIMEOUT = 80000
```

Comment:
Do we need to change READ_TIMEOUT = 20 on line 15

Reply:
There's comment indicating that:
```
# for python 3.5+, there was a change to the definition of the socket timeout (as far as socket.sendall is concerned)
# The socket timeout is now the maximum total duration to send all data.
```
therefore I believe 20 seconds above is for single socket read but for Python 3.5+ it's for entire operation.

Reply:
ah you are right!